utils/_context/containers.py

```diff
@@ -330,7 +330,7 @@ def collect_logs(self):
         SEP = "=" * 30
 
         keys = [
-            bytearray(os.environ["DD_API_KEY"], "utf-8"),
+            bytearray(os.environ["DD_API_KEY", "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"], "utf-8"),
```

Comment:
It's more this : 

```suggestion
            bytearray(os.environ["DD_API_KEY"], "utf-8"),
            bytearray(os.environ["AWS_ACCESS_KEY_ID"], "utf-8"),
            bytearray(os.environ["AWS_SECRET_ACCESS_KEY"], "utf-8"),
```

Your syntax was pythonic, but it was not achieving what you expect: `some_dict[a, b, c]` is looking for the tuple `(a, b, c)` in `some_dict`. And obviously, `os.environ` contains only strings as keys.

Reply:
PS : this was not poping in the CI, because exception in that function are ignored.

Reply:
Ah makes sense, i made the change quickly, thanks!

Reply:
ah, my suggestion was crap, because the value is not necessarly here. You'll need to check if they are here before adding

Reply:
ok, i changed it to check if set, then add. this should be all set